### DISCUSSION TITLE
Property must not be accessed before initialization

### DISCUSSION THREAD

**Discussion by voidnowhere** [livewire/livewire#5603] [1. Help] [answered]
Hi, I have an input text that is wired to a model property named `sub_type`, the value of this input is set using `JS`.
```js
document.getElementById('input').value = some_value;
```
When i submit the form, I notice that the property `sub_type` doesn't get updated. Alternatively if I set the value of the input using keyboard input (manually) the property is updated.

**Comment by joshhanley** [comment, marked as answer]
@voidnowhere changing `value` on an input directly won't be picked up by Livewire. Instead you should use Livewire's javascript api to update the property. Like `@this.sub_type = some_value`. See https://laravel-livewire.com/docs/2.x/inline-scripts#accessing-javascript-component-instance

Hope this helps!

**Reply by voidnowhere** [reply]
Thank you ^_^

**Reply by voidnowhere** [reply]
I noticed that after setting value using the `@this.sub_type = some_value`, the component re renders. Is there a another way to set value without re rendering as an example `wire:model.defer = sub_type`.

**Reply by joshhanley** [reply]
@voidnowhere no worries! 🙂 yeah you can use the set function, `@this.set('sub_type', some_value, true)`. The third parameter `true` tells it to defer. See the full API here (`@this` and `$wire` work the same, just one is Alpine and the other is just javascript) https://laravel-livewire.com/docs/2.x/alpine-js#interacting-with-livewire-from-alpine

**Reply by voidnowhere** [reply]
Thank you ^^

**Reply by joshhanley** [reply]
No worries! 🙂
